listeners/test-client.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+var Mocha = require('mocha');
+var fs = require('fs');
+var path = require('path');
+var Q = require('q');
+
+class TestClient {
+
+    constructor () {
+      var self = this;
+      self._testResult = Q.defer();
+    }
+
+    _getTestResult(){
+      var self = this;
+      return self._testResult.promise;
+    }
+
+    _runTest(asrResult, speakerIds){
+      var self = this;
+
+      //  Get the variables set by main.rule
+      var params = asrResult.NLParse;
+      if (params.action === 'testing') {
+          var dir = '';
+          switch (params.type) {
+              case 'unit':
+                  dir = './test/unit/';
+                  break;
+
+              case 'func':
```

Comment:
change this to 'functional'
I realized if you say 'func' to Jibo it might interpret it as 'funk'